Update ANGLE to 2024-05-13 (decf0e9d4297fafffcacc98b4b9cbc4896962348)

kkinnunen-apple opened the pull request:
#### ae40218113904d2f72cc9b0b9e2c6009d8479f41
<pre>
Update ANGLE to 2024-05-13 (decf0e9d4297fafffcacc98b4b9cbc4896962348)
<a href="https://bugs.webkit.org/show_bug.cgi?id=274080">https://bugs.webkit.org/show_bug.cgi?id=274080</a>
<a href="https://rdar.apple.com/problem/127995658">rdar://problem/127995658</a>

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline 8332e5b10e4f37bf20f63b0c0e8f8ddcae184b17..decf0e9d4297fafffcacc98b4b9cbc4896962348 --pretty=%h %s
decf0e9d42 Roll vulkan-deps from 643408ea2a06 to 7e66c5e2f87e (5 revisions)
4e887491e1 Revert &quot;Document thread-unsafe iterator access to resource maps&quot;
9eaf12d381 Roll Chromium from 4fea14b56dd7 to e8c93a04c96c (1172 revisions)
c96435eda8 Use Chromium&apos;s abseil build files with Dawn.
d1bb6ed839 Document thread-unsafe iterator access to resource maps
8d23fa3f92 Add .gcs_entries to .gitignore
dfddb4f7f4 Roll vulkan-deps from 7c61bb741fb6 to 643408ea2a06 (6 revisions)
4cacc66e70 Remove unused ResourceMap::find
1202055c97 Vulkan: Updates to perf counters
394e8767c6 WebGPU: Fix accumulating mCurrentColorAttachments.
a70ef7fc6e Add EGL_ANGLE_no_error for disabling EGL validation.
2ecfc5a1c2 Issue 8644: Deeply nested iterator declarations can crash
bcb6db5c1a Roll SwiftShader from 76f7f8cfea80 to da334852e705 (5 revisions)
d3b380fc8b Roll vulkan-deps from 44adb2021ffb to 7c61bb741fb6 (6 revisions)
602512a021 Manual roll vulkan-deps from 74fe79763664 to 44adb2021ffb (62 revisions)
36cd4c1f07 Adding basic readPixels.
6a6192c865 Tests: Add native to EGLPrintEGLinfoTest
071e1e83ae EGL: eglCreateSyncKHR uses a return value for the tail call
4c66e8a4ff Roll Chromium from 7a22cfe5a3e7 to 4fea14b56dd7 (650 revisions)
1dc836141f Suppress experimental Linux/Intel failure
1d0ef51841 Fixing glClear tests.
e29d643e52 Vulkan: Fix missing per-present-mode query
3d04180cb5 Vulkan: Add a dedicated garbage list for RefCountedEvents
5b9e7f2043 Vulkan: Release mCurrentEvent before oneOff surface image copy
0905ac0720 Roll third_party/dawn/ ed2c7c713..5406e9dfb (307 commits; 7 trivial rolls)
f3089d1d12 Metal: fix UBO data update undetected between draw calls.
6d10e966db Switch ANGLE to new GPU Android mixins
62eb8fbeec Vulkan: Switch to using zlib::crc32_z for CRC generation
da1fbbc649 Roll Chromium from 4beb6b144a74 to 7a22cfe5a3e7 (616 revisions)
5c78d08420 Android: explicitly disable stack protection
f8f2a10b80 Prepare ANGLE to use new Android mixins
c3a1cae44e Use angle::SimpleMutex everywhere in libGLESv2
a951e0e082 Metal: add test for active blit/compute &amp; render encoder mistracking.
b0875f124a Roll VK-GL-CTS from 17c6e3c4dfad to 9d7b4c3d5533 (11 revisions)
ff0a33cde2 Roll Chromium from dbc600a2b0d5 to 4beb6b144a74 (625 revisions)
465f6b7d2f Android: forward visibility flags
58742dfb79 Use trim() instead of resize()
251067edce Metal: don&apos;t break render pass when uploading a texture.
9ccb9b4102 Metal: fix incorrect tracking if blit &amp; render encoder coexist
28d4c3eb68 Vulkan: Remove BarrierType argument from ImageHelper::barrierImpl
ef8d9f10e6 Vulkan: Use DEVICE_ONLY VkEvent when available
5eb3bca0a4 Vulkan: Minor cleanup
4880a6c207 Enhance VulkanPerformanceCounterTest coverage
d30e47727d Vulkan: Add VkCmdWaitEvents for image barriers
ead98ed94e Implement basic texture upload.
81c2b6e73a Vulkan: Account for padBuffersToMaxVertexAttribStride
9f3e3bff37 Manual roll Chromium from b689a9ee3ba2 to dbc600a2b0d5 (109 revisions)
9aa6d3daf6 Add trim support to MemoryBuffer
be9e63ad09 spirv::Print without ANGLE_ENABLE_ASSERTS -&gt; compile error
97aaad3a1d Vulkan: Pack DescriptorSetLayoutDesc layout
04e469bcc1 Roll Chromium from 904b00ce27b4 to b689a9ee3ba2 (548 revisions)
a96e9197f5 Vulkan: Add RefCountedEvent class and VkCmdSetEvent call
e4a12a676c Vulkan: Dynamic depth test + static depth write
e87122056f End ANGLE Mac Intel experiment
04825894d2 Vulkan: Turn SPIR-V limitations to crash instead of security bug
e560647104 Start Linux/Intel experiment
f2182a713e Manual roll Chromium from bc4f0a08ba8d to 904b00ce27b4 (619 revisions)
44bdc2af94 restricted_trace_perf: Write summary for single renderer
caebfea1f9 Vulkan: Make PipelineBarrierArray a class
1bb1ee061f GL: Sync unpack state for glCompressedTexSubImage3D
53811e86cc Roll Chromium from 83c70f11490d to bc4f0a08ba8d (625 revisions)
bb6a5724e6 Use vendor ID to check QCOM device for tests
950afec46e Vulkan: Add VK_KHR_synchronization2 feature flag
2d1bb3e161 CL: Replace Spinlock with angle::SimpleMutex
6658d5e8e5 Tests: Add Modern Combat 5 trace
298b87398d Vulkan: Restrict the ContextVk dependency in CommandBufferHelper
e49b941fef Roll Chromium from 228731187f89 to 83c70f11490d (641 revisions)
b839e05ba7 GLSL: Hardcode emulated atan functions
1064b38d8d Vulkan: Move ImageLayout define to earlier in vk_helpers.h file
4929e1df3e Vulkan: Add feature flag for using event for image barrier
f51f2bc55b Manual roll vulkan-deps from 736920240f0b to 74fe79763664 (13 revisions)
038ad6c755 EGL: Use unlocked tail calls for surface, sync and img calls
978cf07b73 Suppress failures on Pixel 6 AP1A.240405.002
44ce58871c Allow the backend to do resource init for framebuffers.
a0b721bc70 Vulkan: Optimize DescriptorSetLayout with no bindings
2eed792ca6 Vulkan: Use angle::SimpleMutex with async command queue
acb8b33f40 Vulkan: Enable 8/16 bit storage buffer extensions
5907fcfd1f Roll Chromium from 0129806a25c3 to 228731187f89 (630 revisions)

Canonical link: <a href="https://commits.webkit.org/278744@main">https://commits.webkit.org/278744@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/057d98c5dd801b9c393043f3bfceb5a8b6105f8e

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/51305 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/30611 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/3647 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/54565 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/1995 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/36922 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/1674 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/41783 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/53404 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/28251 "Passed tests") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/18/builds/44309 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/22901 "Passed tests") | 
| | [❌ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/42/builds/25576 "4 new passes 10 flakes") | [❌ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/1491 "Found 60 new test failures: accessibility/ARIA-reflection.html, accessibility/accessibility-crash-with-dynamic-inline-content.html, accessibility/accessibility-node-reparent.html, accessibility/display-contents/aria-grid.html, accessibility/display-contents/aria-hidden.html, accessibility/mac/abbr-acronym-tags.html, accessibility/mac/accesskey.html, accessibility/mac/attributed-string/attributed-string-does-not-includes-misspelled-for-non-editable.html, accessibility/mac/attributed-string/attributed-string-for-range-with-options.html, accessibility/mac/search-text/search-text.html ... (failure)") | [⏳ 🛠 wpe-skia ](https://ews-build.webkit.org/#/builders/WPE-Skia-Build-EWS "Waiting in queue, processing has not started yet") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/47545 "Passed tests") | [❌ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/1564 "Found 60 new test failures: accessibility/ARIA-reflection.html, accessibility/combobox/aria-combobox-control-owns-elements.html, accessibility/combobox/aria-combobox-hierarchy.html, accessibility/combobox/aria-combobox-no-owns.html, accessibility/combobox/aria-combobox.html, accessibility/combobox/combobox-active-element-selected-children.html, accessibility/combobox/combobox-linked-listbox-destroyed.html, accessibility/combobox/mac/aria-combobox-activedescendant.html, accessibility/combobox/mac/combobox-activedescendant-notifications.html, accessibility/combobox/mac/combobox-inherited-role.html ... (failure)") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/56158 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/26420 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/62/builds/1460 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/49181 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/27663 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/44296 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/48345 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11253 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/28553 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/27398 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

djg reviewed (commented):
Looks like unrelated test failures.

:shipit:

kkinnunen-apple committed:
Update ANGLE to 2024-05-13 (decf0e9d4297fafffcacc98b4b9cbc4896962348)
https://bugs.webkit.org/show_bug.cgi?id=274080
rdar://problem/127995658

Reviewed by Dan Glastonbury

Contains upstream commits:
git log --oneline 8332e5b10e4f37bf20f63b0c0e8f8ddcae184b17..decf0e9d4297fafffcacc98b4b9cbc4896962348 --pretty=%h %s
decf0e9d42 Roll vulkan-deps from 643408ea2a06 to 7e66c5e2f87e (5 revisions)
4e887491e1 Revert "Document thread-unsafe iterator access to resource maps"
9eaf12d381 Roll Chromium from 4fea14b56dd7 to e8c93a04c96c (1172 revisions)
c96435eda8 Use Chromium's abseil build files with Dawn.
d1bb6ed839 Document thread-unsafe iterator access to resource maps
8d23fa3f92 Add .gcs_entries to .gitignore
dfddb4f7f4 Roll vulkan-deps from 7c61bb741fb6 to 643408ea2a06 (6 revisions)
4cacc66e70 Remove unused ResourceMap::find
1202055c97 Vulkan: Updates to perf counters
394e8767c6 WebGPU: Fix accumulating mCurrentColorAttachments.
a70ef7fc6e Add EGL_ANGLE_no_error for disabling EGL validation.
2ecfc5a1c2 Issue 8644: Deeply nested iterator declarations can crash
bcb6db5c1a Roll SwiftShader from 76f7f8cfea80 to da334852e705 (5 revisions)
d3b380fc8b Roll vulkan-deps from 44adb2021ffb to 7c61bb741fb6 (6 revisions)
602512a021 Manual roll vulkan-deps from 74fe79763664 to 44adb2021ffb (62 revisions)
36cd4c1f07 Adding basic readPixels.
6a6192c865 Tests: Add native to EGLPrintEGLinfoTest
071e1e83ae EGL: eglCreateSyncKHR uses a return value for the tail call
4c66e8a4ff Roll Chromium from 7a22cfe5a3e7 to 4fea14b56dd7 (650 revisions)
1dc836141f Suppress experimental Linux/Intel failure
1d0ef51841 Fixing glClear tests.
e29d643e52 Vulkan: Fix missing per-present-mode query
3d04180cb5 Vulkan: Add a dedicated garbage list for RefCountedEvents
5b9e7f2043 Vulkan: Release mCurrentEvent before oneOff surface image copy
0905ac0720 Roll third_party/dawn/ ed2c7c713..5406e9dfb (307 commits; 7 trivial rolls)
f3089d1d12 Metal: fix UBO data update undetected between draw calls.
6d10e966db Switch ANGLE to new GPU Android mixins
62eb8fbeec Vulkan: Switch to using zlib::crc32_z for CRC generation
da1fbbc649 Roll Chromium from 4beb6b144a74 to 7a22cfe5a3e7 (616 revisions)
5c78d08420 Android: explicitly disable stack protection
f8f2a10b80 Prepare ANGLE to use new Android mixins
c3a1cae44e Use angle::SimpleMutex everywhere in libGLESv2
a951e0e082 Metal: add test for active blit/compute & render encoder mistracking.
b0875f124a Roll VK-GL-CTS from 17c6e3c4dfad to 9d7b4c3d5533 (11 revisions)
ff0a33cde2 Roll Chromium from dbc600a2b0d5 to 4beb6b144a74 (625 revisions)
465f6b7d2f Android: forward visibility flags
58742dfb79 Use trim() instead of resize()
251067edce Metal: don't break render pass when uploading a texture.
9ccb9b4102 Metal: fix incorrect tracking if blit & render encoder coexist
28d4c3eb68 Vulkan: Remove BarrierType argument from ImageHelper::barrierImpl
ef8d9f10e6 Vulkan: Use DEVICE_ONLY VkEvent when available
5eb3bca0a4 Vulkan: Minor cleanup
4880a6c207 Enhance VulkanPerformanceCounterTest coverage
d30e47727d Vulkan: Add VkCmdWaitEvents for image barriers
ead98ed94e Implement basic texture upload.
81c2b6e73a Vulkan: Account for padBuffersToMaxVertexAttribStride
9f3e3bff37 Manual roll Chromium from b689a9ee3ba2 to dbc600a2b0d5 (109 revisions)
9aa6d3daf6 Add trim support to MemoryBuffer
be9e63ad09 spirv::Print without ANGLE_ENABLE_ASSERTS -> compile error
97aaad3a1d Vulkan: Pack DescriptorSetLayoutDesc layout
04e469bcc1 Roll Chromium from 904b00ce27b4 to b689a9ee3ba2 (548 revisions)
a96e9197f5 Vulkan: Add RefCountedEvent class and VkCmdSetEvent call
e4a12a676c Vulkan: Dynamic depth test + static depth write
e87122056f End ANGLE Mac Intel experiment
04825894d2 Vulkan: Turn SPIR-V limitations to crash instead of security bug
e560647104 Start Linux/Intel experiment
f2182a713e Manual roll Chromium from bc4f0a08ba8d to 904b00ce27b4 (619 revisions)
44bdc2af94 restricted_trace_perf: Write summary for single renderer
caebfea1f9 Vulkan: Make PipelineBarrierArray a class
1bb1ee061f GL: Sync unpack state for glCompressedTexSubImage3D
53811e86cc Roll Chromium from 83c70f11490d to bc4f0a08ba8d (625 revisions)
bb6a5724e6 Use vendor ID to check QCOM device for tests
950afec46e Vulkan: Add VK_KHR_synchronization2 feature flag
2d1bb3e161 CL: Replace Spinlock with angle::SimpleMutex
6658d5e8e5 Tests: Add Modern Combat 5 trace
298b87398d Vulkan: Restrict the ContextVk dependency in CommandBufferHelper
e49b941fef Roll Chromium from 228731187f89 to 83c70f11490d (641 revisions)
b839e05ba7 GLSL: Hardcode emulated atan functions
1064b38d8d Vulkan: Move ImageLayout define to earlier in vk_helpers.h file
4929e1df3e Vulkan: Add feature flag for using event for image barrier
f51f2bc55b Manual roll vulkan-deps from 736920240f0b to 74fe79763664 (13 revisions)
038ad6c755 EGL: Use unlocked tail calls for surface, sync and img calls
978cf07b73 Suppress failures on Pixel 6 AP1A.240405.002
44ce58871c Allow the backend to do resource init for framebuffers.
a0b721bc70 Vulkan: Optimize DescriptorSetLayout with no bindings
2eed792ca6 Vulkan: Use angle::SimpleMutex with async command queue
acb8b33f40 Vulkan: Enable 8/16 bit storage buffer extensions
5907fcfd1f Roll Chromium from 0129806a25c3 to 228731187f89 (630 revisions)

Canonical link: https://commits.webkit.org/278744@main
